lib/spack/spack/build_systems/autotools.py

```diff
@@ -196,20 +196,21 @@ def _do_patch_libtool(self):
         detect the compiler (and patch_libtool is set), patch in the correct
         flags for the Arm, Clang/Flang, and Fujitsu compilers."""
 
-        libtool = os.path.join(self.build_directory, "libtool")
-        if self.patch_libtool and os.path.exists(libtool):
-            if self.spec.satisfies('%arm') or self.spec.satisfies('%clang') \
-                    or self.spec.satisfies('%fj'):
-                for line in fileinput.input(libtool, inplace=True):
-                    # Replace missing flags with those for Arm/Clang
-                    if line == 'wl=""\n':
-                        line = 'wl="-Wl,"\n'
-                    if line == 'pic_flag=""\n':
-                        line = 'pic_flag="{0}"\n'\
-                               .format(self.compiler.cc_pic_flag)
-                    if self.spec.satisfies('%fj') and 'fjhpctag.o' in line:
-                        line = re.sub(r'/\S*/fjhpctag.o', '', line)
-                    sys.stdout.write(line)
+        for libtool in find(self.build_directory, 'libtool', recursive=True):
+            if self.patch_libtool:
+                if self.spec.satisfies('%arm')\
```

Comment:
Can we extract the part that does the actual patching into its own method or function? In this way this method would read:
```python
def _do_patch_libtool(self):
    if not self.pacth_libtool:
        return
    
    for libtool_path in find(self.build_directory, 'libtool', recursive=True):
        _patch_libtool(libtool_path)
```

Reply:
I fixed it.